lib/src/analyzer_impl.dart

```diff
@@ -122,8 +125,13 @@ class AnalyzerImpl {
   void prepareErrors() {
     for (Source source in sources) {
       context.computeErrors(source);
+
       var sourceErrors = context.getErrors(source);
       errorInfos.add(sourceErrors);
+
+      if (options.strongMode) {
+        errorInfos.add(strongChecker.computeErrors(source));
+      }
```

Comment:
Yeah, it's a valid concern and does happen in some cases now. The messages should for the most part be extra messages about things Analyzer does not report. However, warnings do get upgraded to errors in at least some cases, I noticed some around invalid assignment and overrides. DDC CodeChecker and OverrideChecker classes (sort of like StaticTypeAnalyzer and ErrorVerifier in analyzer) can issue duplicate messages, but errors instead of warnings.

Personally I think we should integrate even more tightly. In an ideally world, DDC's strong subtype rules plug into existing resolution (e.g. StaticTypeAnalyzer), which would make stronger checks just "fall out" of existing Analyzer passes. It would also reduce code duplication/maintenance in the long run. I think our lesson from integrating inference more tightly was that it worked great, so we should do the same thing for the DDC visitor currently called CodeChecker. On top of these benefits, errors would only be emitted once.

DDC would ideally like to upgrade warnings to errors. In other words, "strong mode" implies warnings as errors.